### DISCUSSION TITLE
Getting a String Value from the Web Browser Problem.

### DISCUSSION THREAD

**Discussion by dcipher** [webui-dev/webui#372] [General]
Hi

I am using Purebasic with webui and everything works except one strange case. Now it maybe that I am doing this wrong but the following codes works as long as I send something other than zeros. I am trying to send values back to my Purebasic app from the web browser.  Basically I am capturing user input from the web page. Below are some examples values sent back to the Purebasic app indicating which ones were successfully captured.  The ones that contain just zeros appear as empty strings in Purebasic.  That would make sense if I was send back 0x00 but I am sending back 0x30. Is this a bug or a mistake on my part?

javascript:
```js
 webui.call("SetParam",'Cancel'); // This works
 webui.call("SetParam",'1'); // This works
 webui.call("SetParam",'0'); // This does not work as I get an empty string.
 webui.call("SetParam",'0000'); // This does not work as I get an empty string.
 webui.call("SetParam",'0200'); // This works
```        
Purebasic:

```purebasic
ProcedureC SetParam(*e.webui_event_t) 
   goParam\Input = PeekS(webui_get_string(*e),-1,#PB_Ascii)
   If goParam\Input = "Cancel"
      goParam\Cancelled = 1
   EndIf
EndProcedure

goParam\webUI = webui_new_window()
webui_bind(goParam\webUI,"SetParam",@SetParam())
```

Thanks,
Simon

**Comment by hassandraga** [comment]
Understood. To quickly identify the issue, we may need the logs.
Please replace the WebUI library with [the nightly](https://github.com/webui-dev/webui/releases/tag/nightly) **debug** build, then share the logs.

**Reply by dcipher** [reply]
Where are the logs located?

**Reply by dcipher** [reply]
I downloaded the nightly build and ram make but is that how you get a debug build?  I apologize as I am not familiar with the build system etc. Perhaps that is why I do not see any logs?

**Reply by hassandraga** [reply]
You can build the debug version from source by running `make debug` on Linux, or `nmake debug` on Windows. However, the nightly build is already compiled and ready to use.

1. Download the nightly build depend on your OS and CPU architecture
2. Copy WebUI library `.../debug/webui-2.dll` if Windows, or `.../debug/webui-2.so` if Linux
3. Paste and replace the already existing WebUI library file in your Purebasic
4. Run your Purebasic in a terminal to see the logs

```
C:\test.exe

[User] webui_new_window()
[User] webui_get_new_window_id()
[Core]          WebUI v2.5.0-Beta-1 (Microsoft Windows, Non-Encrypted)       
[Core]          _webui_init()
...
```

**Reply by dcipher** [reply]
Hi

The reason I build it from scratch is that OpenSuse does not have glibc_2.33 and glibc_2.34 so I get linker errors. Opensuse has glibc_2.31.
